code/modules/reagents/machinery/electrolyzer.dm

```diff
@@ -0,0 +1,346 @@
+/obj/machinery/electrolyzer
+	name = "electrolyzer"
+	density = TRUE
+	anchored = TRUE
+	icon = 'icons/obj/machines/chemistry.dmi'
+	icon_state = "electrolysis"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 40
+	circuit = /obj/item/weapon/circuitboard/electrolyzer
+	var/obj/item/weapon/reagent_containers/beaker
+	var/obj/item/weapon/reagent_containers/separationBeaker
+	var/convertion_coefficient = 2
+	var/on = FALSE
+
+/obj/item/weapon/circuitboard/electrolyzer
+	name = T_BOARD("electrolyzer")
+	build_path = /obj/machinery/electrolyzer
+	origin_tech = list(TECH_BIO = 3)
+
+// returns FALSE on errors TRUE on success and -1 if nothing to do
+/proc/electrolysis(var/obj/item/weapon/reagent_containers/primaryBeaker, var/obj/item/weapon/reagent_containers/secondaryBeaker, var/amount)
+	if(!primaryBeaker || !secondaryBeaker)
+		return FALSE
+	//check if has reagents
+	if(!primaryBeaker.reagents.total_volume)
+		return FALSE
+
+	//check only one reagent present
+	if(!primaryBeaker.reagents.reagent_list.len > 1)
+		return FALSE
+
+	var/datum/chemical_reaction/originalReaction
+	var/activeReagent
+	for(var/datum/reagent/R in primaryBeaker.reagents.reagent_list)
+		for(var/id in GLOB.chemical_reactions_list_by_result)
```

Comment:
the loop isn't needed, just access it directly and check if recipeList isn't null